### ISSUE TITLE
clone calibration tag displays twice; clear tags before refreshing

### ISSUE THREAD

**Issue by jywarren** [publiclab/spectral-workbench#217]
and write a test for this

Also, seeing very serious load times for /spectrums/choose/calibration:

````sql
^[  Spectrum Load (179887.4ms)  SELECT DISTINCT spectrums.* FROM `spectrums` INNER JOIN `tags` ON `tags`.`spectrum_id` = `spectrums`.`id` WHERE (tags.name = ('calibration')) ORDER BY id DESC LIMIT 6 OFFSET 0
  EXPLAIN (158.1ms)  EXPLAIN SELECT DISTINCT spectrums.* FROM `spectrums` INNER JOIN `tags` ON `tags`.`spectrum_id` = `spectrums`.`id` WHERE (tags.name = ('calibration')) ORDER BY id DESC LIMIT 6 OFFSET 0
EXPLAIN for: SELECT  DISTINCT spectrums.* FROM `spectrums` INNER JOIN `tags` ON `tags`.`spectrum_id` = `spectrums`.`id` WHERE (tags.name = ('calibration')) ORDER BY id DESC LIMIT 6 OFFSET 0
+----+-------------+-----------+--------+--------------------+--------------------+---------+-----------------------------------------------+-------+----------------------------------------------+
| id | select_type | table     | type   | possible_keys      | key                | key_len | ref                                           | rows  | Extra                                        |
+----+-------------+-----------+--------+--------------------+--------------------+---------+-----------------------------------------------+-------+----------------------------------------------+
|  1 | SIMPLE      | tags      | ref    | index_tags_on_name | index_tags_on_name | 258     | const                                         | 26356 | Using where; Using temporary; Using filesort |
|  1 | SIMPLE      | spectrums | eq_ref | PRIMARY            | PRIMARY            | 4       | spectralworkbench_production.tags.spectrum_id |     1 |                                              |
+----+-------------+-----------+--------+--------------------+--------------------+---------+-----------------------------------------------+-------+----------------------------------------------+
2 rows in set (0.16 sec)
````

This query doesn't seem so bad:

````sql
SELECT DISTINCT spectrums.* FROM `spectrums` INNER JOIN `tags` ON `tags`.`spectrum_id` = `spectrums`.`id` WHERE (tags.name = ('calibration')) ORDER BY id DESC LIMIT 6 OFFSET 0
````

But yikes, 180 seconds. Need an index at least? @btbonval -- any input on this one? I assume we're primarily dealing with some really gigantic tables. I'll start by not selecting the data column.

**Comment by jywarren** [comment]
Did this and writing a test for the choose method in spectrums_controller.rb. Committing and testing.

**Comment by jywarren** [comment]
Optimization solved! As of be0493e
